cmake/FindSol2.cmake

```diff
@@ -11,10 +11,14 @@ if (Sol2_FOUND)
     )
     target_compile_definitions(Sol2 INTERFACE 
         SOL_ALL_SAFETIES_ON=1
-        SOL_USING_CXX_LUA=1
+        SOL_USING_CXX_LUA=0
```

Comment:
This likely needs to be conditional based on what lua version we detect. But evidently there is some non_trivial source chaanges needed as well based om the test failures that bult lua from scratch (although that one should still be building the cxx version: https://github.com/Chatterino/chatterino2/blob/115428d7f475c75c0ae6c211f0db4c81004ce49b/lib/lua/CMakeLists.txt)